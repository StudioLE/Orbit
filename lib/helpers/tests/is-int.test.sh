#!/bin/bash
set -euo pipefail

# ARGS
ACTUAL_FILE=$1

source /srv/lib/helpers/is-int.sh

declare OUTPUT=""

for i in -1 0 1 10 100000 0.5 -0.5 1.0 .0 "" 1
do
  OUTPUT+="$i: "
  OUTPUT+=$(is-int "$i")
  OUTPUT+=$'\n'
done

OUTPUT+="NO ARGS: "
set +e
OUTPUT+=$(is-int 2>&1)
EXIT_CODE=$?
set -e
OUTPUT+=" EXIT_CODE: ${EXIT_CODE}"
OUTPUT+=$'\n'

echo "${OUTPUT}" > "${ACTUAL_FILE}"
