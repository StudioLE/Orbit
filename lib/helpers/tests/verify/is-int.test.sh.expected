-1: TRUE
0: TRUE
1: TRUE
10: TRUE
100000: TRUE
0.5: 
-0.5: 
1.0: 
.0: 
: 
1: TRUE
NO ARGS: ‚ùó  $1 was not set. EXIT_CODE: 1

